unstructured/partition/html/parser.py

```diff
@@ -472,6 +617,152 @@ def _iter_text_segment(self, emphasis: str) -> Iterator[TextSegment]:
         if text := self.text:
             yield TextSegment(text, self._annotation(text, emphasis))
 
+    def _iter_text_segments_from_block_tail_and_phrasing(
+        self, tail: str, q: deque[Flow | Phrasing], emphasis: str
+    ) -> Iterator[TextSegment | Element]:
+        """Generate zero-or-more `TextSegment`s or `Element`s from tail+phrasing of block child.
+
+        When this phrasing element contains a block child (not valid HTML but accepted by
+        browsers), the tail of that block child and any phrasing elements contiguous with that tail
+        also need to contribute their text. This method takes care of that job.
+
+        Note this mutates `q` by popping phrasing elements off as they are processed.
+        """
+        if tail:
+            yield TextSegment(tail, self._annotation(tail, emphasis))
+        while q and q[0].is_phrasing:
+            e = cast(Phrasing, q.popleft())
+            yield from e.iter_text_segments(emphasis)
+
+
+class Anchor(Phrasing):
+    """Custom element-class for `<a>` element.
+
+    Provides link annotations.
+    """
+
+    def iter_text_segments(self, enclosing_emphasis: str = "") -> Iterator[TextSegment | Element]:
+        """Generate text segments for contents and tail of this element, when they exist.
+
+        Phrasing is emitted as `TextSegment` objects. Any nested block items (not valid HTML but
+        are accepted by browser so can occur) are emitted as `Element` objects.
+
+        When an anchor contains a nested block element, there can be multiple phrases and/or
+        elements. Link annotation is only added to the first phrase or element. Otherwise the link
+        annotation would span multiple document-elements.
+        """
+        q: deque[Phrase | Element] = deque(self._iter_phrases_and_elements(enclosing_emphasis))
+
+        # -- the first non-whitespace phrase or element gets the link annotation --
+        while q:
+            print(f"{q=}")
```

Comment:
Testing out in the API - this print is pretty noisy.

Reply:
Oooh, that's not supposed to be there! Good catch, thanks @awalker4 :)